### PR TITLE
Added DAO/repository layer for doctor entity

### PR DESCRIPTION
**Pull request by SChandrakanth15**
created the Entity class for Doctor and Done with the repository layer.

### CODE REVIEW

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.theelixrlabs.healthcare.constants;
+
+public class DoctorConstants {
+    public static final String DOCTOR_COLLECTION_NAME = "doctor";
```
Comment (ashutosh-2609) [inline src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java]: is collection specific to single doctor ?
if not change it to "doctors"

File: src/main/java/com/theelixrlabs/healthcare/model/Doctor.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.util.UUID;
+
+@Data
+@Document(collection = DoctorConstants.DOCTOR_COLLECTION_NAME)
+public class Doctor {
+    @Id
+    private String uuid = UUID.randomUUID().toString();
```
Comment (ashutosh-2609) [inline src/main/java/com/theelixrlabs/healthcare/model/Doctor.java]: 1. Assigning the id value here is not recommended 
2. Also UUID can be used in place of String
3. variable name uuid is not correct. UUID is a type which represents id value so keep the name id itself

File: src/main/java/com/theelixrlabs/healthcare/model/Doctor.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import com.theelixrlabs.healthcare.constants.DoctorConstants;
+import lombok.Data;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.util.UUID;
+
+@Data
+@Document(collection = DoctorConstants.DOCTOR_COLLECTION_NAME)
+public class Doctor {
```
Comment (ashutosh-2609) [inline src/main/java/com/theelixrlabs/healthcare/model/Doctor.java]: You can name the classes as per their usage
DoctorsModel ar DoctorsEntity

Also add some class level comments

**Review by ashutosh-2609 — CHANGES_REQUESTED** [review]
Please check on comments

**Review by vinodkadam22 — COMMENTED** [review]
Add class level comments for each class
Every one maintain same DB name

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java (hunk 1 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java]: yes changed it to doctors.

File: src/main/java/com/theelixrlabs/healthcare/model/Doctor.java (hunk 2 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/model/Doctor.java]: 1.done
2.have doubt on this.
3.done

File: src/main/java/com/theelixrlabs/healthcare/model/Doctor.java (hunk 3 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/model/Doctor.java]: the name changed to "DoctorsEntity". and aslo added some class level comments.

File: src/main/java/com/theelixrlabs/healthcare/model/DoctorsEntity.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Entity class representing doctors in the healthcare system.
+ * This class is mapped to the MongoDB collection specified by DoctorConstants.DOCTOR_COLLECTION_NAME.
+ */
+@Data
+@Document(collection = DoctorConstants.DOCTOR_COLLECTION_NAME)
+public class DoctorsEntity {
+    @Id
+    private UUID id;
+    private String firstName;
```
Comment (EXR-38) [inline src/main/java/com/theelixrlabs/healthcare/model/DoctorsEntity.java:23]: Are these fields nullable? It is a good practice to constraint validations to these fields.

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.theelixrlabs.healthcare.constants;
+
+/**
+ * Constants related to doctors in the healthcare system.
+ */
+public class DoctorConstants {
+    public static final String DOCTOR_COLLECTION_NAME = "doctors";
```
Comment (EXR-38) [inline src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java]:
```suggestion
    public static final String DOCTORS_COLLECTION_NAME = "doctors";
```

File: src/main/java/com/theelixrlabs/healthcare/model/DoctorsEntity.java (hunk 4 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/model/DoctorsEntity.java:23]: Done added the validations for the fields.

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.theelixrlabs.healthcare.constants;
+
+/**
+ * Constants related to doctors in the healthcare system.
+ */
+public class DoctorConstants {
+    public static final String DOCTORS_COLLECTION_NAME = "doctors";
+    public static final String FIRST_NAME_SHOULD_BE_MANDATORY = "First name is mandatory";
```
Comment (EXR-38) [inline src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java]: Better to move the messages here to a message.properties file under the resources folder.

File: build.gradle [hunk 7]
```diff
@@ -25,9 +25,10 @@ repositories {
 }
 
 dependencies {
-	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
 	implementation 'org.springframework.boot:spring-boot-starter-data-mongodb'
 	implementation 'org.springframework.boot:spring-boot-starter-web'
+	implementation 'javax.validation:validation-api:2.0.1.Final'
+	implementation 'org.hibernate.validator:hibernate-validator:6.2.0.Final'
```
Comment (EXR-38) [inline build.gradle]: Is this required here?

**Review by EXR-38 — CHANGES_REQUESTED** [review]

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java (hunk 5 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java]: Changed

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java (hunk 6 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/constants/DoctorConstants.java]: moved the messages to a separate MessageConstants class.

File: build.gradle (hunk 7 above)
Comment (SChandrakanth15) [reply build.gradle]: removed the hibernate. validator dependency.

**Review by EXR-38 — APPROVED** [review]

**Review by divya-sathyan — APPROVED** [review]

**Review by ashutosh-2609 — APPROVED** [review]

**Review by vinodkadam22 — APPROVED** [review]

**Pull request merged**